src/main/scala/org/apache/spark/deploy/armada/submit/ArmadaClientApplication.scala

```diff
@@ -264,73 +509,147 @@ private[spark] class ArmadaClientApplication extends SparkApplication {
 
     val executorCount = SchedulerBackendUtils.getInitialTargetExecutorNumber(conf)
 
+    // Resolve values using precedence function
+    val finalQueue = resolveRequiredValue(
+      armadaJobConfig.cliConfig.queue.filter(_.nonEmpty),
+      armadaJobConfig.jobTemplate.map(_.queue).filter(_.nonEmpty),
+      s"Queue name must be set via ${ARMADA_JOB_QUEUE.key} or in the job template."
+    )
+
+    val finalJobSetId = armadaJobConfig.cliConfig.jobSetId match {
+      case Some(id) if id.nonEmpty => id
+      case Some(_) =>
+        throw new IllegalArgumentException(
+          s"Empty jobSetId is not allowed. Please set a valid jobSetId via ${ARMADA_JOB_SET_ID.key}"
+        )
+      case None =>
+        armadaJobConfig.jobTemplate.map(_.jobSetId).filter(_.nonEmpty).getOrElse(conf.getAppId)
+    }
+
```

Comment:
It's a bit awkward that we are repeating a bunch of the code from validateRequiredConfig() here.  Is there any way we could unify it.  I think I understand why they are separate, but it forces the reader to think about why we need it in two places.